SpatialGDK/Documentation/content/singleton-actors.md

```diff
@@ -1,55 +1,70 @@
+       GENERATED_BODY()
+       ...
+     }
+     ``` 
 
-The code snippet below shows how to tag a native C++ class with the appropriate Private Singleton identifiers.
+### Unreal Blueprint Class
 
-```
-UCLASS(SpatialType=(Singleton, ServerOnly))
-class TESTSUITE_API AExampleGameGameMode : public AGameModeBase
-{
-  GENERATED_BODY()
-  ...
-}
-```
+1. Open your Blueprint Class in the [Blueprint Editor](https://docs.unrealengine.com/en-us/Engine/Blueprints/Editor).
+2. From the [Toolbar](https://docs.unrealengine.com/en-US/Engine/Blueprints/Editor/UIComponents/Toolbar), click **Class Settings** to open Blueprint Properties in the Default pane.
+3. In the **Class Options** section, ensure that the **Spatial Type** check box is selected.
+4. In the **Spatial Description** text box, tag the Blueprint Class with the relevant attribute(s), depending on what type of Singleton Actor you want it to be:
+   * **Public Singleton Actor**: enter `Singleton`.
+   * **Private Singleton Actor**: enter `Singleton ServerOnly`.
+  Here's a screenshot of what your Blueprint Class looks like if you tag it as a Private Singleton Actor:
 
-To tag a Blueprint class as a Public Singleton, open the class in the Blueprint Editor and navigate to the `Class Settings`. In the `Advanced` section inside `Class Options`, check the `Spatial Type` checkbox and add `Singleton` to the `Spatial Description` textbox. To tag a Blueprint class as a Private Singleton, follow the same steps and add `ServerOnly` to the `Spatial Description` textbox.
-
-This is an example of what your Blueprint `Class Options` should look like if you've tagged it as a Private Singleton:
 ![Singleton Blueprint]({{assetRoot}}assets/screen-grabs/blueprint-singleton.png)
 
-And that's it! You have successfully specified a Singleton Actor. Make sure that if you edit your Actor, you then generate [schema]({{urlRoot}}/content/spatialos-concepts/schema-and-snapshots#schema) and create a new [snapshot]({{urlRoot}}/content/how-to-use-snapshots) using the [SpatialOS GDK toolbar]({{urlRoot}}/content/toolbars).
\ No newline at end of file
+## Managing Singleton Actors
+
+Because each server-worker instance spawns each Singleton Actor locally, we have introduced the [Global State Manager]({{urlRoot}}/content/glossary#global-state-manager) (GSM) to ensure that there is only ever one instance of an [entity]({{urlRoot}}/content/glossary#entity) that represents a Singleton Actor, and to ensure data is replicated properly. 
+
+The GSM allows only the server-worker instance with authority over the GSM to spawn a representative entity for each Singleton Actor. Every other server-worker instance spawns each Singleton Actor locally and then links it to the one on the server-worker instance that has authority over the GSM. Additionally, each Public Singleton Actor is replicated to every client-worker instance via the normal [Actor replication flow](https://docs.unrealengine.com/en-us/Gameplay/Networking/Actors/ReplicationFlow).
```

Comment:
```suggestion
The GSM allows only the server-worker instance with [authority]({{urlRoot}}/content/glossary#authority) over the GSM to spawn a representative entity for each Singleton Actor. Every other server-worker instance spawns each Singleton Actor locally and then links it to the one on the server-worker instance that has authority over the GSM. Additionally, each Public Singleton Actor is replicated to every client-worker instance via the normal [Actor replication flow](https://docs.unrealengine.com/en-us/Gameplay/Networking/Actors/ReplicationFlow).
```

Reply:
I just doubt that users do want/need to know what's authority when they arrive here [with a certain level of knowledge].

Reply:
It's a fair point, and I know we've discussed it before. It's possible that we tend to go over the top with our cross-linking in the docs!